From 96d74b4121dfd604f24631282ce1e9b409091fef Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 04 Sep 2020 12:31:40 +0300
Subject: [PATCH] [SW-12754] gaudi: security: disable hbw rrs

For the HBW RRs to be really disabled, it is not enough to zero the base
and mask data, but we also need to switch the polarity, so "hit" will
miss base and mask of zeros.
Finally disable the HBW RRs.

Change-Id: Ic67d9451995d40fcdd3825dd645979e4e592111c
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index e662da4..d32e514 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -840,6 +840,8 @@
 
 static void preboot_security_pci_disable(void)
 {
+	u32 hit;
+
 	regmap_writel(0, mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MIN_0);
 	regmap_writel(0, mmPCIE_WRAP_RR_WR_PRV_RANGE_REG_MAX_0);
 	regmap_writel(0, mmPCIE_WRAP_RR_RD_PRV_RANGE_REG_MIN_0);
@@ -853,11 +855,19 @@
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_BASE_HIGH_AW_0);
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_MASK_LOW_AW_0);
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_MASK_HIGH_AW_0);
+	/* PCIe RTR RR - Disable (hit) RR */
+	hit = regmap_readl(mmSIF_RTR_CTRL_7_RANGE_PRIV_HIT_AW);
+	regmap_writel(hit & ~BIT(0), mmSIF_RTR_CTRL_7_RANGE_PRIV_HIT_AW);
+
 	/* PCIe RTR RR - HBM RP 0x0 - 0x07FFFFFFFF */
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_BASE_LOW_AR_0);
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_BASE_HIGH_AR_0);
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_MASK_LOW_AR_0);
 	regmap_writel(0, mmSIF_RTR_CTRL_7_RANGE_PRIV_MASK_HIGH_AR_0);
+	/* PCIe RTR RR - Disable (hit) RR */
+	hit = regmap_readl(mmSIF_RTR_CTRL_7_RANGE_PRIV_HIT_AR);
+	regmap_writel(hit & ~BIT(0), mmSIF_RTR_CTRL_7_RANGE_PRIV_HIT_AR);
+
 	preboot_puts("Disabled SIF RRs\n");
 }
 
